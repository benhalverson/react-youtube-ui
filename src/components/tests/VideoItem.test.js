import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react'
import VideoItem from '../VideoItem'

Enzyme.configure({ adapter: new Adapter() });

describe('Component: VideoItem', () => {
  
  test('should render', () => {
    const video = {
      snippet: {
        thumbnails: {
          medium: {
            url: 'imageurl.jpg'
          }
        },
        title: 'Hello world'
      }
    }
    const wrapper = Enzyme.shallow(<VideoItem video={ video } />)

    expect(wrapper.render()).toMatchSnapshot();
  });

  test('should trigger event on select', () => {
    const video = {
      snippet: {
        thumbnails: {
          medium: {
            url: 'imageurl.jpg'
          }
        },
        title: 'Hello world'
      }
    }
    const spyEvent = jest.fn()
    const wrapper = Enzyme.shallow(<VideoItem video={ video } onVideoSelect={ spyEvent } />)
    
    // trigger click
    wrapper.find('.video-item').simulate('click')

    expect(spyEvent).toBeCalledWith(video)
  })

});