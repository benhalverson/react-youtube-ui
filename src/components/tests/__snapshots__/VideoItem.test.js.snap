// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: VideoItem should render 1`] = `
"<div className=\\"video-item item\\" onClick={[Function: onClick]}>
  <img className=\\"ui image\\" src=\\"imageurl.jpg\\" alt=\\"Hello world\\" />
  <div className=\\"content\\">
    <div className=\\"header\\">
      Hello world
    </div>
  </div>
</div>"
`;
